source/common/network/utility.cc

```diff
@@ -335,6 +341,21 @@ bool Utility::isLoopbackAddress(const Address::Instance& address) {
   NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
 }
 
+bool Utility::isSameIPAddress(const Address::Instance& address1,
+                              const Address::Instance& address2) {
+  if (address1.type() != Address::Type::Ip ||
+      address1.ip()->version() != address2.ip()->version()) {
+    return false;
+  }
+
+  if (address1.ip()->version() == Address::IpVersion::v4) {
```

Comment:
This might look less verbose as a switch statement.

Reply:
Done.